### PR TITLE
[FTAMS102020] Alejandro Pliakas Benitez

### PR DESCRIPTION
**Pull request by apliakas**

### CODE REVIEW

File: routes/drones.js [hunk 1]
```diff
@@ -1,37 +1,57 @@
...
 const router = express.Router();
 
 router.get('/drones', (req, res, next) => {
-  // Iteration #2: List the drones
-  // ... your code here
+  Drone.find({})
+    .then((DronesFromDB) => {
+      res.render("drones/list", { DronesFromDB })
+    })
+    .catch((error) => `Error while fetching all drones: ${error}`);
```
Comment (LloydC) [inline routes/drones.js:12]: You're missing `console.log()` or `console.error()` to display the error.

File: routes/drones.js [hunk 2]
```diff
@@ -1,37 +1,57 @@
...
 
 router.post('/drones/create', (req, res, next) => {
-  // Iteration #3: Add a new drone
-  // ... your code here
+  const { name, propellers, maxSpeed } = req.body;
+  console.log({ name, propellers, maxSpeed });
+
+  Drone.create({ name, propellers, maxSpeed },)
+    .then(() => res.redirect('/drones'))
+    .catch((error) => `Error while creating drone: ${error}`);
```
Comment (LloydC) [inline routes/drones.js:26]: Same comment as above

File: routes/drones.js [hunk 3]
```diff
@@ -1,37 +1,57 @@
...
-  // Iteration #4: Update the drone
-  // ... your code here
+  const { id } = req.params;
+
+  Drone.findById(id)
+    .then((drone) => {
+      console.log(drone);
+      res.render('drones/update-form', drone);
+    })
+    .catch((error) => `Error while editing drone: ${error}`);
```
Comment (LloydC) [inline routes/drones.js:37]: Same comment as above

File: routes/drones.js [hunk 4]
```diff
@@ -1,37 +1,57 @@
...
 
 router.post('/drones/:id/edit', (req, res, next) => {
-  // Iteration #4: Update the drone
-  // ... your code here
+  const { id } = req.params;
+  const { name, propellers, maxSpeed } = req.body
+
+  Drone.findByIdAndUpdate(id, { name, propellers, maxSpeed }, { new: true })
+    .then(() => res.redirect('/drones'))
+    .catch((error) => `Error while editing drone: ${error}`);
```
Comment (LloydC) [inline routes/drones.js:46]: Same comment as above

File: routes/drones.js [hunk 5]
```diff
@@ -1,37 +1,57 @@
...
 });
 
 router.post('/drones/:id/delete', (req, res, next) => {
-  // Iteration #5: Delete the drone
-  // ... your code here
+  const { id } = req.params;
+
+  Drone.findByIdAndDelete(id)
+  .then(() => res.redirect('/drones'))
+  .catch((error) => `Error while deleting drone: ${error}`);
```
Comment (LloydC) [inline routes/drones.js:54]: Same comment as above

File: views/drones/create-form.hbs [hunk 6]
```diff
@@ -1 +1,16 @@
-<h2>Create a new drone</h2>
\ No newline at end of file
+<div>
+  <h2>Create a new drone</h2>
+
+  <form action='create' method='post'>
```
Comment (LloydC) [inline views/drones/create-form.hbs:4]: The path of the action should be `/drones/create` instead of `create`

File: views/drones/list.hbs [hunk 7]
```diff
@@ -1 +1,32 @@
...
+
+  <div class="drones">
+    {{#each DronesFromDB}}
+
+    <div class="dronebox">
+      
+      <h3>{{name}}</h3>
+      <div class="middledronebox">Number of propellers: {{propellers}}. </div>
+      <div class="speed-drone" >Max. speed: {{maxSpeed}} m per second.</div>
+      <form action="drones/{{_id}}/edit" method="GET">
```
Comment (LloydC) [inline views/drones/list.hbs:15]: You could also use an anchor element `<a href ="/drones/{{_id}}/edit"> Edit drone</a>`, you should only use a form if you're looking to pass data along your GET request

File: views/drones/list.hbs [hunk 8]
```diff
@@ -1 +1,32 @@
...
+      <form class='deletebttn' action="drones/{{_id}}/delete" method="post">
+        <button type="submit">Delete drone</button>
+      </form>
+      
+    </div>
+
+    {{/each}}
+  </div>
+
+  <a class='createdrone' href="/drones/create"> Create a drone </a>
```
Comment (LloydC) [inline views/drones/list.hbs:27]: I recommend using camelCase notation when you create a class made of two words `createDrone` (it's best practice)

Comment (apliakas) [reply views/drones/list.hbs:25]: fixed the console.logs, the path, I changed the form for an anchor (I left i as a button for visual purposes, it was easier than modifying an anchor to look like another button) and I will be more careful with the names, I just pushed it to my forked repository. Thanks for the review! :)

**Pull request closed without merging**